### ISSUE TITLE
contentContext creates "undefined" object wrapper around objects in an array

### ISSUE THREAD

**Issue by masyukun** [marklogic/marklogic-data-hub#1652]
#### The issue

Short description of the problem:
When harmonizing, I want to assign an array of objects into the returned content object:
```javascript
  let email = xdmp.quote( !fn.empty(fn.head(source.xpath('//EmailAddress'))) 
    ? [].concat( fn.head(fn.head(source.xpath('//EmailAddress'))).toString().split("|").map(function(email){
         return { "emailAddress": email }
        })  )
      : null
  );

  // return the instance object
  return {
    '$attachments': attachments,
    '$type': 'Persona',
    '$version': '0.0.1',
    'id': id,
    'name': name,
    'email': email,
    'phone': phone
  }
```
From this code, I expect (and see returned when I run in Query console:

```json
{
	"envelope": {
		"headers": {},
		"triples": {},
		"instance": {
			"Persona": {
				"id": 234,
				"name": "Jim Driscoll",
				"email": [
					{
						"emailAddress": "jim.driscoll@marklogic.com"
					}
				],
				"phone": "555-123-4567"
			},
			"info": {
				"title": "Persona",
				"version": "0.0.1"
			}
		}
	}
}
```

Instead, I see:

```json
{
	"envelope": {
		"headers": {},
		"triples": {},
		"instance": {
			"Persona": {
				"id": 234,
				"name": "Jim Driscoll",
				"email": [
					{
						"undefined": {
							"emailAddress": "jim.driscoll@marklogic.com"
						}
					}
				],
				"phone": "555-123-4567"
			},
			"info": {
				"title": "Persona",
				"version": "0.0.1"
			}
		}
	}
}
```
What behavior are you expecting?
No wrapping {"undefined": XXX} object

#### Tech details

Which Operating System are you using? Any

Which version of MarkLogic are you using? 9.0-6

Which version of the Data Hub Framework are you using? 4.0.1

Is this a QuickStart UI bug? If so, which browser are you using? No

#### The devil is in the details...

If possible, rerun the command with -d for debugging output and attach the output:

  **for quickstart:**  
  `java -jar quickstart.war -d > myoutput.txt`

  **for gradle:**  
  `gradle mlYourCommand -d > myoutput.txt`

**Comment by masyukun** [comment]
I know there is another issue related to this, but the outcome seems to be "this is how it's intended to work." https://github.com/marklogic/marklogic-data-hub/issues/1302

If I could suggest, rather than the undefined" structure, just default to not wrapping the object.

**Comment by masyukun** [comment]
Current workaround is xdmp.quote() in content.sjs, then xdmp.unquote() in writer.sjs -- clearly not a great solution.

**Comment by ayuwono** [comment]
Tracking this on JIRA: https://project.marklogic.com/jira/browse/DHFPROD-1641

**Comment by srinathgit** [comment]
@masyukun ,

1. Can you please provide the entity definition file for the entity 'Persona' (Persona.entity.json)  ?
2. Does the following email data adhere to the "email" property definition specified in the 'Persona' entity ?
"email": [
					{
						"emailAddress": "jim.driscoll@marklogic.com"
					}
				]

**Comment by srinathgit** [comment]
@masyukun ,

The expectation in harmonization flow is that the value of  "email" adheres to what is being specified in the entity 'Persona'. It can be scalar("email": "jim.driscoll@marklogic.com"), array ("email":["jim.driscoll@marklogic.com"]) or a reference

**Comment by ayuwono** [comment]
Fixed on 4.1.1

**Comment by netdance** [comment]
I'm still seeing this behavior.
